app/affirmation/page.tsx

```diff
@@ -0,0 +1,133 @@
+"use client";
+import React, { useState, useEffect } from 'react';
+import Header from '../header';
+import Footer from '../footer';
+import Modal from 'react-modal';
+import { affirmations } from '../state';
+
+interface Button {
+  id: number;
+  text: string;
+}
+
+function Home() {
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [currentEditId, setCurrentEditId] = useState<number | null>(null);
+  const [editText, setEditText] = useState('');
+  const initialButtons: Button[] = [
+    { id: 1, text: 'Say what I want to say, whatever happens will help me grow' },
+    { id: 2, text: 'I can take up space' },
+    { id: 3, text: 'I have an important voice' },
+    { id: 4, text: 'Feel the tension and proceed' },
+    { id: 5, text: 'I have the right to stutter' },
+  ];
+
+  // Initialize state with saved data if it exists, otherwise use the initial data.
+  const [buttons, setButtons] = useState<Button[]>(() => {
+    const stringifiedAffirmations = affirmations.getAffirmationsAsString();
+    return stringifiedAffirmations ? JSON.parse(stringifiedAffirmations) : initialButtons;
+  });
+
+  useEffect(() => {
+    affirmations.setAffirmationsAsString(JSON.stringify(buttons));
+  }, [buttons]);
+
+  const openModal = (button: Button) => {
+    setModalIsOpen(true);
+    setCurrentEditId(button.id);
+    setEditText(button.text);
+  };
+
+  const closeModal = () => {
+    setModalIsOpen(false);
+  };
+
+  const handleEdit = () => {
+    setButtons((prevButtons) =>
+      prevButtons.map((button) =>
+        button.id === currentEditId ? { ...button, text: editText } : button
+      )
+    );
+    console.log(buttons);
+    closeModal();
+  };
+
+  const deleteButton = (id: number) => {
+    setButtons((prevButtons) => prevButtons.filter((button) => button.id !== id));
+  };
+
+  const addButton = () => {
+    const newId = Math.max(0, ...buttons.map((b) => b.id)) + 1; // Create a new ID
+    const newButton: Button = { id: newId, text: '' };
+    setButtons((prevButtons) => [...prevButtons, newButton]);
+    setModalIsOpen(true);
+    setCurrentEditId(newId);
+  };
+
+  const saveAffirmation = (text: string) => {
+    affirmations.setCurrentAffirmation(text);
+  };
+
+  return (
+    <div className="bg-white w-screen h-screen">
+      <div className="flex w-full justify-between">
+        <Header />
+      </div>
+
+      <div style={{ marginLeft: '20px', marginRight: '20px' }}>
```

Comment:
In general, I think most of this styling would be more natural to put in a CSS file rather than sprinkled throughout the TSX code.  That applies doubly to styling that is repeated, e.g for each button.

Reply:
@kexinF, you can also have a generic component TSX code (e.g. button, checkbox) that is reusable. I found some good examples here:

https://github.com/nextauthjs/next-auth-example/tree/main/components